modules/dag-l1/src/main/scala/org/tessellation/dag/l1/domain/consensus/block/BlockConsensusInput.scala

```diff
@@ -13,39 +13,38 @@ import org.tessellation.security.signature.signature.Signature
 import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}
 import io.circe.{Decoder, Encoder}
 
-sealed trait BlockConsensusInput[+T <: Transaction] extends Ω
+sealed trait BlockConsensusInput extends Ω
 
 object BlockConsensusInput {
-  sealed trait OwnerBlockConsensusInput extends BlockConsensusInput[Nothing]
+  sealed trait OwnerBlockConsensusInput extends BlockConsensusInput
   case object OwnRoundTrigger extends OwnerBlockConsensusInput
   case object InspectionTrigger extends OwnerBlockConsensusInput
 
-  sealed trait PeerBlockConsensusInput[+T <: Transaction] extends BlockConsensusInput[T] {
+  sealed trait PeerBlockConsensusInput extends BlockConsensusInput {
     val senderId: PeerId
     val owner: PeerId
   }
 
   object PeerBlockConsensusInput {
-    implicit def encoder[T <: Transaction: Encoder]: Encoder[PeerBlockConsensusInput[T]] = deriveEncoder
-    implicit def decoder[T <: Transaction: Decoder]: Decoder[PeerBlockConsensusInput[T]] = deriveDecoder
+    implicit def encoder: Encoder[PeerBlockConsensusInput] = deriveEncoder
```

Comment:
I think you can derive it with derevo right now.

Reply:
:+1: